addons/cookoff/functions/fnc_isMagazineFlare.sqf

```diff
@@ -0,0 +1,28 @@
+#include "script_component.hpp"
+/*
+ * Author: Cyruz
+ * Checks if the magazine has ammo which is a flare
+ *
+ * Arguments:
+ * 0: Magazine <STRING>
+ *
+ * Return Value:
+ * 0: If magazine is type of flare <BOOL>
+ *
+ * Example:
+ * ["3Rnd_UGL_FlareWhite_F"] call ace_cookoff_fnc_isMagazineFlare
+ *
+ * Public: No
+ */
+
+params ["_magazine"];
+
+private _ammo = getText (configFile >> "CfgMagazines" >> _magazine >> "ammo");
+private _intensity = getNumber (configFile >> "CfgAmmo" >> _ammo >> "intensity");
+private _flare = getNumber (configFile >> "CfgAmmo" >> _ammo >> QEGVAR(grenades,flare));
+
+if (_intensity != 0 || _flare == 1) then {
+    true
+} else {
+    false
+};
```

Comment:
Put just the condition as the last line (don't need the `if`/`else`), and use lazy eval.

Reply:
Adjusted, not using lazy here as you have some cases where flare is 1 but intensity is 0. I.e. ACE hand flares.

Reply:
I'm pretty sure lazy eval will be slower in case of simple number comparison anyway.